src/team_lib/mhba/xccl_mhba_collective.c

```diff
@@ -48,10 +47,16 @@ static xccl_status_t xccl_mhba_node_fanin(xccl_mhba_team_t     *team,
             ctrl_v = (int *)((ptrdiff_t)team->node.ops[index].ctrl +
                              MHBA_CTRL_SIZE * i);
             if (*ctrl_v != request->seq_num) {
```

Comment:
i would re-write this multi-condition like this:
```
if (*ctrl_v == -request->seq_num) {
    request->buff_change_flag = 1;
} else if (*ctrl_v != request->seq_num) {
    return XCCL_INPROGRESS;
}
```
It has less code and more readable imho.